src/pymor/solvers/interface.py

```diff
@@ -0,0 +1,96 @@
+# This file is part of the pyMOR project (https://www.pymor.org).
+# Copyright pyMOR developers and contributors. All rights reserved.
+# License: BSD 2-Clause License (https://opensource.org/licenses/BSD-2-Clause)
+
+from pymor.core.base import ImmutableObject
+from pymor.core.exceptions import LinAlgError
+
+
+class Solver(ImmutableObject):
+    """Equation solver.
+
+    Solves equations of the form `AU = V` for `U`.
+    """
+
+    least_squares = False
+    jacobian_solver = None
+
+    @property
+    def adjoint_solver(self):
+        return self
+
+    def solve(self, operator, V, mu=None, initial_guess=None, return_info=False):
+        assert self.least_squares or operator.source.dim == operator.range.dim
+        assert V in operator.range
+        assert initial_guess is None or initial_guess in operator.source and len(initial_guess) == len(V)
+
+        # always treat handle the zero dimensional case here to avoid having to do this for every
+        # linear solver implementation
```

Comment:
```suggestion
        # always treat the zero-dimensional case here to avoid having to do this for every
        # linear solver implementation
```